src/overlay005/encounter_effect.c

```diff
@@ -0,0 +1,1673 @@
+
+    if (screenFlash->done != NULL) {
+        *(screenFlash->done) = FALSE;
+    }
+
+    screenFlash->screen = screen;
+    screenFlash->screenFlashColor = screenFlashColor;
+    screenFlash->otherScreenFlashColor = otherScreenFlashColor;
+    screenFlash->numFlashes = numFlashes;
+}
+
+static void EncounterEffect_ExecuteFlash(SysTask *task, void *param)
+{
+    ScreenFlash *screenFlash = param;
+
+    switch (screenFlash->state) {
+    case SCREENFLASH_STATE_INIT_OTHER_SCREEN_FADE:
+        if (screenFlash->screen == SCREEN_TOP) {
+            BrightnessFadeTask_Init(&screenFlash->otherScreenFadeTask, 0, screenFlash->otherScreenFlashColor, 2, 8);
+        } else {
+            if (screenFlash->screen == SCREEN_BOTTOM) {
+                BrightnessFadeTask_Init(&screenFlash->otherScreenFadeTask, 0, screenFlash->otherScreenFlashColor, 1, 8);
+            }
+        }
+
+        screenFlash->state++;
+        break;
+    case SCREENFLASH_STATE_TARGET_INIT_SCREEN_FADE:
+        BrightnessFadeTask_Init(&screenFlash->screenFadeTask, 0, screenFlash->screenFlashColor, screenFlash->screen, 3);
+        screenFlash->state++;
+        break;
+    case SCREENFLASH_STATE_WAIT_FOR_TARGET_SCREEN_FADE:
+        if (BrightnessFadeTask_Update(&screenFlash->screenFadeTask)) {
+            screenFlash->state++;
+        }
+        break;
+    case SCREENFLASH_STATE_RESET_TARGET_SCREEN_FADE:
+        BrightnessFadeTask_Init(&screenFlash->screenFadeTask, screenFlash->screenFlashColor, 0, screenFlash->screen, 3);
+        screenFlash->state++;
+        break;
+    case SCREENFLASH_STATE_WAIT_FOR_TARGET_SCREEN_FADE_RESET:
+        if (BrightnessFadeTask_Update(&screenFlash->screenFadeTask)) {
+            screenFlash->flashCounter++;
+
+            if (screenFlash->flashCounter == screenFlash->numFlashes) {
+                screenFlash->state = SCREENFLASH_STATE_FINISH;
+            } else {
+                screenFlash->state = SCREENFLASH_STATE_TARGET_INIT_SCREEN_FADE;
+            }
+        }
+        break;
+    case SCREENFLASH_STATE_FINISH:
+        screenFlash->state = SCREENFLASH_STATE_INIT_OTHER_SCREEN_FADE;
+        screenFlash->flashCounter = 0;
+
+        if (screenFlash->done != NULL) {
+            *(screenFlash->done) = TRUE;
+        }
+
+        SysTask_Done(task);
+        Heap_FreeToHeap(screenFlash);
+
+        return;
+    }
+
+    BrightnessFadeTask_Update(&screenFlash->otherScreenFadeTask);
+}
+
+BOOL EncounterEffect_GetHBlankFlag(EncounterEffect *param0)
+{
+    return param0->hBlankFlag;
+}
+
+void LinearInterpolationTaskS32_Init(LinearInterpolationTaskS32 *task, int start, int end, int numSteps)
+{
+    task->currentValue = start;
+    task->startValue = start;
+    task->delta = end - start;
+    task->numSteps = numSteps;
+    task->currentStep = 0;
+}
+
+BOOL LinearInterpolationTaskS32_Update(LinearInterpolationTaskS32 *task)
+{
+    int interpolated = task->delta *task->currentStep;
+    interpolated = interpolated / task->numSteps;
```

Comment:
1. put a space between both sides of a binary operator (here, the `*`)
2. does this match as the following?

```suggestion
    int interpolated = (task->delta * task->currentStep) / task->numSteps;
```